src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -105,9 +166,12 @@ private void getMealById(HttpServletRequest request, HttpServletResponse respons
             response.sendError(HttpServletResponse.SC_NOT_FOUND);
             return;
         }
-
-        Meal meal = dishes.get(id);
-
+        Filter propertyFilter = new FilterPredicate("id", FilterOperator.EQUAL, id);
+        Query query = new Query("Meal").setFilter(propertyFilter);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        ArrayList<Meal> meals = getDataFromDatastore(results); 
+        Meal meal = meals.get(0);
```

Comment:
Be prepared for a case where number of meals returned by Datastore is not equal to 1.
If the number is 0, then a "404 NOT FOUND" error should be prepared.
If the number is higher than 1, then something went very wrong - you may want to abort and prepare a "500 INTERNAL SERVER ERROR".